h2/src/main/org/h2/mvstore/MVStore.java

```diff
@@ -404,28 +407,36 @@ private static void compact(MVStore source, MVStore target) {
                     targetMeta.put(key, m.getValue());
                 }
             }
-            // We are going to cheat a little bit in the copyFrom() by employing "incomplete" pages,
-            // which would be spared of saving, but save completed pages underneath,
-            // and those may appear as dead (non-reachable).
-            // That's why it is important to preserve all chunks
-            // created in the process, especially if retention time
-            // is set to a lower value, or even 0.
-            for (String mapName : source.getMapNames()) {
-                MVMap.Builder<Object, Object> mp = MVStoreTool.getGenericMapBuilder();
-                // This is a hack to preserve chunks occupancy rate accounting.
-                // It exposes design deficiency flaw in MVStore related to lack of
-                // map's type metadata.
-                // TODO: Introduce type metadata which will allow to open any store
-                // TODO: without prior knowledge of keys / values types and map implementation
-                // TODO: (MVMap vs MVRTreeMap, regular vs. singleWriter etc.)
-                if (mapName.startsWith(TransactionStore.UNDO_LOG_NAME_PREFIX)) {
-                    mp.singleWriter();
-                }
-                MVMap<Object, Object> sourceMap = source.openMap(mapName, mp);
-                MVMap<Object, Object> targetMap = target.openMap(mapName, mp);
-                targetMap.copyFrom(sourceMap);
-                targetMeta.put(MVMap.getMapKey(targetMap.getId()), sourceMeta.get(MVMap.getMapKey(sourceMap.getId())));
-            }
+
+            int poolSize = Integer.getInteger("h2.compactThreads", 1);
```

Comment:
Why would we not want to use half or quarter of the available cores as default instead?